src/MediatR/Mediator.cs

```diff
@@ -58,7 +58,7 @@ public Task<TResponse> Send<TResponse>(IRequest<TResponse> request, Cancellation
 
                     if (!isValidRequest)
                     {
-                        throw new ArgumentException($"{nameof(request)} does not implement ${nameof(IRequest)}");
+                        throw new ArgumentException($"{requestType.Name} does not implement {nameof(IRequest)}");

```

Comment:
i also found a bug with the exception message

Reply:
Should we add the argument name here? `nameof(requestType)`?

Reply:
done